### PR TITLE
feat: add actions to .github repo

### PR DESCRIPTION
**Pull request by NickAzureDevops**
## Description

copy actions from recalls-infra repo in the CICD branch to the .github location 

Related issue: [1279](https://dvsa.atlassian.net/browse/MOTRECALLS-1279)

## Before submitting (or marking as "ready for review")

- [x ] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [ x] Have you performed a self-review of the code
- [ ] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

**Comment by AnnaDodson** [comment]
Looks good!

Do you have some links to show these actions running? Are they working with workflows in subfolders?

Fantastic adding the READMEs. Please can you also add:
- Add links to them from the main README so people can find them easily
- Add version numbers to the examples in the READMEs instead of pointing `@main` so if there's breaking changes, everyones workflows won't break

**Comment by AnnaDodson** [comment]
> @AnnaDodson Thank you for the review, we will work on your suggestions. For version number, would you be wanting the actions to point to 3.1.0 or 4.0.0.

@jcrozier - 3.2.0 please! Then we will make a new release once your changes are in

File: .github/actions/build-name/action.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    value: ${{ steps.branch.outputs.NAME }}
+  artifact_prefix:
+    value: ${{ steps.branch.outputs.NAME }}-${{ steps.sha.outputs.SHORT_SHA }}
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v3
+    - name: Set short sha output
+      id: sha
+      run: echo "SHORT_SHA=$(git rev-parse --short=11 HEAD)" >> $GITHUB_OUTPUT
```
Comment (AnnaDodson) [inline .github/actions/build-name/action.yml:15]: Hi @NickAzureDevops, @JoshuaLicense has rewritten this action to get the pretty branch name. I suggest you remove this actions from this PR and use the one Josh will be adding shortly instead.

Comment (NickAzureDevops) [reply .github/actions/build-name/action.yml]: @AnnaDodson  Done. Removed it from the root of the readme.md so @JoshuaLicense  will have to add it again once he added his action

**Review by AnnaDodson — APPROVED** [review]
Approved. I've chased for any other comments

**Pull request merged**